pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,359 +17,172 @@
-	// +nullable
-	Snapshots *bool `json:"snapshots,omitempty"`
-}
-
-// AccessMode of the volumes created on pool
-type AccessMode string
+// StoragePoolConditionType specifies the particular state that the conditions are based on.
+type StoragePoolConditionType string
 
 const (
-	// ReadWriteOnce can be mounted in read/write mode to exactly 1 host
-	ReadWriteOnce AccessMode = "ReadWriteOnce"
+	// StoragePoolConditionTypePoolExpansion condition will be available when user triggers
+	// pool expansion by adding disks to the pool
+	StoragePoolConditionTypePoolExpansion StoragePoolConditionType = "PoolExpansion"
 
-	// ReadOnlyMany can be mounted in read-only mode to many hosts
-	ReadOnlyMany AccessMode = "ReadOnlyMany"
+	// StoragePoolConditionTypeDiskReplacement condition will be available when user triggers
+	// disk replacement.
+	StoragePoolConditionTypeDiskReplacement StoragePoolConditionType = "DiskReplacement"
 
-	// ReadWriteMany can be mounted in read/write mode to many hosts
-	ReadWriteMany AccessMode = "ReadWriteMany"
-)
+	// StoragePoolConditionTypeDiskUnavailable condition will be available when one (or) more
+	// disks were unavailable
+	StoragePoolConditionTypeDiskUnavailable StoragePoolConditionType = "DiskUnavailable"
 
-// ProvisioningPolicy for the pool
-type ProvisioningPolicy string
-
-const (
-	// ThickProvisioning for thick provisioning
-	ThickProvisioning ProvisioningPolicy = "thick"
-
-	// ThinProvisioning for thin provisioning
-	ThinProvisioning ProvisioningPolicy = "thin"
-)
-
-// StoragePoolMultipathingOption supported by the pool
-type StoragePoolMultipathingOption string
-
-const (
-	OnlineActiveMultipathingOption  StoragePoolMultipathingOption = "OnlineActive"
-	OnlinePassiveMultipathingOption StoragePoolMultipathingOption = "OnlinePassive"
-	NoneMultipathingOption          StoragePoolMultipathingOption = "None"
-)
-
-// CompressionAlgorithm supported by the pool
-type CompressionAlgorithm string
-
-// MediaEncryptionAlgorithm supported by the pool
-type MediaEncryptionAlgorithm string
-
-// DataSanitizationPolicy supported by the pool
-type DataSanitizationPolicy string
-
-const (
-	NoneDataSanitizationPolicy               DataSanitizationPolicy = "None"
-	ClearDataSanitizationPolicy              DataSanitizationPolicy = "Clear"
-	CryptographicEraseDataSanitizationPolicy DataSanitizationPolicy = "CryptographicErase"
-)
-
-// AccessProtocol supported by the pool
-type AccessProtocol string
-
-const (
-	NVMeAccessProtocol           AccessProtocol = "NVMe"
-	NVMeOverFabicsAccessProtocol AccessProtocol = "NVMeOverFabics"
-)
-
-// StoragePoolTier for the pool
-type StoragePoolTier string
-
-const (
-	PlatinumTier StoragePoolTier = "P1"
-	GoldTier     StoragePoolTier = "P2"
-	SilverTier   StoragePoolTier = "P3"
+	// StoragePoolConditionTypelPoolLost condition will be available when the underlying specific pool
+	// is not in usable.
+	StoragePoolConditionTypelPoolLost StoragePoolConditionType = "PoolLost"
```

Comment:
Is it better to have "PoolHealthy" and "PoolOffline" as well?
We can replace "PoolLost" with "PoolOffline" .